pkg/controller/trustbundle-publisher/controller.go

```diff
@@ -0,0 +1,206 @@
+
+	// Fetch the configmap
+	addlBundlecfgMap := &corev1.ConfigMap{}
+	if err := r.client.Get(context.TODO(), names.AddlTrustBundleName(), addlBundlecfgMap); err != nil {
+		if apierrors.IsNotFound(err) {
+			// This means the configmap was already deleted and there are
+			// stale queue entries (or something edge triggering from a related
+			// resource that got deleted async).
+			log.Printf("configmap %s/%s not found; reconciliation will be skipped",
+				request.Namespace, request.Name)
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get configmap %q: %v", request, err)
+	}
+
+	additionalData, err := r.validateTrustBundleConfigMap(addlBundlecfgMap)
+	if err != nil {
+		log.Printf("Failed to validate ConfigMap: %v", err)
+		r.status.SetDegraded(statusmanager.TrustBundle, "InvalidConfigMapData",
+			fmt.Sprintf("The configmap data is invalid (%v). Use 'oc edit cm/%s -n %s' to fix.",
+				err, names.ADDL_TRUST_BUNDLE_CONFIGMAP, names.TRUST_BUNDLE_CONFIGMAP_NS))
+		return reconcile.Result{}, nil
+	}
+
+	systemData, err := r.validateSystemTrustBundle(names.SYSTEM_TRUST_BUNDLE)
+	if err != nil {
+		log.Printf("Failed to ensure system trust bundle %q: %v", names.SYSTEM_TRUST_BUNDLE, err)
+		r.status.SetDegraded(statusmanager.TrustBundle, "InvalidSystemTrustBundle",
+			fmt.Sprintf("Invalid system trust bundle %q: %v", names.SYSTEM_TRUST_BUNDLE, err))
+		return reconcile.Result{}, nil
+	}
+
+	mergedCfgMap, err := r.ensureMergedConfigMap(additionalData, systemData)
+	if err != nil {
+		log.Printf("Failed to ensure merged trust bundle configmap %s/%s: %v", mergedCfgMap.Namespace,
+			mergedCfgMap.Name, err)
+		r.status.SetDegraded(statusmanager.TrustBundle, "FailedToEnsureMergedConfigMap",
+			fmt.Sprintf("Merged trust bundle configmap %s/%s is invalid: %v", mergedCfgMap.Namespace,
+				mergedCfgMap.Name, err))
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.syncMergedConfigMap(mergedCfgMap); err != nil {
+		log.Printf("Could not sync merged trust bundle configmap %s/%s: %v", mergedCfgMap.Namespace,
+			mergedCfgMap.Name, err)
+		r.status.SetDegraded(statusmanager.TrustBundle, "FailedToSyncMergedConfigMap",
+			fmt.Sprintf("Could not sync merged trust bundle configmap %s/%s: %v", mergedCfgMap.Namespace,
+				mergedCfgMap.Name, err))
+		return reconcile.Result{}, err
+	}
+	r.status.SetNotDegraded(statusmanager.TrustBundle)
+
+	log.Printf("Completed reconciling %s/%s\n", request.Namespace, request.Name)
+	return reconcile.Result{}, nil
+}
+
+// ensureMergedConfigMap merges the additionalData with systemData into
+// a single byte slice, ensures the merged byte slice contains valid PEM
+// encoded certificates, embeds the merged byte slice into a configmap
+// named "proxy-ca-bundle" in namespace "openshift-config-managed" and
+// returns the configmap.
+func (r *ConfigMapReconciler) ensureMergedConfigMap(additionalData, systemData []byte) (*corev1.ConfigMap, error) {
```

Comment:
Is it a problem if the additional CA is already present in the system trust? Will that break expectations?

Reply:
it's just going to result in 2 copies of that cert in the bundle, right?

Reply:
Yup. The question is: does that break anything? Does the bundle need to be unique? (I have no idea)

Reply:
i'd be surprised but it would definitely be a good thing to test.